Sources/CoreMetrics/Metrics.swift

```diff
@@ -382,6 +382,22 @@ public enum MetricsSystem {
     fileprivate static var _factory: MetricsFactory = NOOPMetricsHandler.instance
     fileprivate static var initialized = false
 
+    /// Acquire a writer lock for the duration of the given block.
+    ///
+    /// - Parameter body: The block to execute while holding the lock.
+    /// - Returns: The value returned by the block.
+    public static func withWriterLock<T>(_ body: () throws -> T) rethrows -> T {
+        return try self.lock.withWriterLock(body)
+    }
+
+    /// Acquire a reader lock for the duration of the given block.
+    ///
+    /// - Parameter body: The block to execute while holding the lock.
+    /// - Returns: The value returned by the block.
+    public static func withReaderLock<T>(_ body: () throws -> T) rethrows -> T {
+        return try self.lock.withReaderLock(body)
+    }
```

Comment:
We don't actually need the reader I think -- just the writer for bootstrap right?

Reply:
For the functionality in #61 we only need the writer, indeed. However I figured for completeness sake and to keep options open I'd add both.

Fine with removing the reader lock as well though 😄